renderers/terminal/clay_renderer_terminal_ansi.c

```diff
@@ -0,0 +1,195 @@
+#include "stdint.h"
+#include "string.h"
+#include "stdio.h"
+#include "stdlib.h"
+#include "math.h"
+
+#ifdef CLAY_OVERFLOW_TRAP
+#include "signal.h"
+#endif
+
+static inline void Console_MoveCursor(int x, int y) {
+    printf("\033[%d;%dH", y + 1, x + 1);
+}
+
+bool Clay_PointIsInsideRect(Clay_Vector2 point, Clay_BoundingBox rect) {
+    // TODO this function is a copy of Clay__PointIsInsideRect but that one is internal, I don't know if we want
+    // TODO to expose Clay__PointIsInsideRect
+    return point.x >= rect.x && point.x < rect.x + rect.width && point.y >= rect.y && point.y < rect.y + rect.height;
+}
+
+static inline void Console_DrawRectangle(int x0, int y0, int width, int height, Clay_Color color,
+                                         Clay_BoundingBox scissorBox) {
+    float average = (color.r + color.g + color.b + color.a) / 4 / 255;
+
+    for (int y = y0; y < height + y0; y++) {
+        for (int x = x0; x < width + x0; x++) {
+            if (!Clay_PointIsInsideRect((Clay_Vector2) {.x = x, .y = y}, scissorBox)) {
+                continue;
+            }
+
+            Console_MoveCursor(x, y);
+            // TODO there are only two colors actually drawn, the background and white
+            if (average > 0.75) {
```

Comment:
This is wrong, color perception is logarithmic on the intensity.

Reply:
I'm sure it is haha, I just threw something in there to get the example looking ok, we can follow this up with a nicer approach 🙂